contributions/MPI_IS_gaussian_process/src/gaussian_process_guider.cpp

```diff
@@ -274,7 +274,17 @@ double GaussianProcessGuider::result(double input, double SNR, double time_step,
         {
             dithering_active_ = false;
         }
-        deduceResult(time_step); // just pretend we would do dark guiding...
+        try
+        {
+            deduceResult(time_step); // just pretend we would do dark guiding...
+        }
+        catch (std::runtime_error err)
+        {
+            reset();
+            std::string outStr = "PPEC: Model reset after exception: " + std::string(err.what());
+            GPDebug->Log(outStr.c_str());

```

Comment:
Log takes a printf-style format string so this could be a little dangerous (I'm thinking of the case where outStr contains a `%` character somewhere)
Here's is a safer way that also gets rid of the unneeded temporary string construction:
```suggestion
            GPDebug->Log("PPEC: Model reset after exception: %s", err.what());
```